rust/kernel/src/sync.rs

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use core::cell::UnsafeCell;
+use core::ops::{Deref, DerefMut};
+use core::fmt;
+
+use crate::bindings;
+
+#[cfg(CONFIG_DEBUG_SPINLOCK)]
+compile_error!("CONFIG_DEBUG_SPINLOCK not yet implemented");
+
+#[cfg(CONFIG_DEBUG_MUTEX)]
+compile_error!("CONFIG_DEBUG_MUTEX not yet implemented");
+
+pub struct Mutex<T> {
+    lock: bindings::mutex,
+    data: UnsafeCell<T>,
+}
+
```

Comment:
Surely this needs to `impl Sync` (and `Send`)?

Reply:
probaply. Never implemented one of them. will try.